conans/client/tools/pkg_config.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import subprocess
+
+_global_output = None
+
+
+def cmd_output(command):
+    return subprocess.check_output(command).decode().strip()
+
+
+class PkgConfig(object):
+
+    def __init__(self, library, pkg_config_executable='pkg-config', static=False, msvc_syntax=False, variables=None):
+        """
+        :param library: library (package) name, such as libastral
+        :param pkg_config_executable: specify custom pkg-config executable (e.g. for cross-compilation)
+        :param static: output libraries suitable for static linking (adds --static to pkg-config command line)
+        :param msvc_syntax: MSVC compatibility (adds --msvc-syntax to pkg-config command line)
+        :param variables: dictionary of pkg-config variables (passed as --define-variable=VARIABLENAME=VARIABLEVALUE)
+        """
+        self.library = library
+        self.pkg_config_executable = pkg_config_executable
+        self.static = static
+        self.msvc_syntax = msvc_syntax
+        self.define_variables = variables
+
+        variable_names = self._parse_output('print-variables').split()
+        self._variables = dict()
+        for name in variable_names:
+            self.variables[name] = self._parse_output('variable=%s' % name)
+
+        self.info = dict()
+        for option in ['cflags', 'cflags-only-I', 'cflags-only-other',
+                       'libs', 'libs-only-L', 'libs-only-l', 'libs-only-other',
+                       'print-provides', 'print-requires', 'print-requires-private']:
+            self.info[option] = self._parse_output(option).split()
```

Comment:
Probably is a better idea to not invoke all the possible options and just to it lazy in the @property methods.
Something like:
```
@property
def cflags(self):
    if not "cflags" in self.info:
         # Call the command and assign to self.info
    return self.info['cflags']
```

Reply:
yep, good suggestion